### PR TITLE
Implemented Karger's Algorithm for finding global mincut

### PR DESCRIPTION
**Pull request by barkincavdaroglu**
This probabilistic algorithm finds the global mincut of a graph by contracting an edge until the order of graph is smaller than or equal to 2. The sum of weights of edges between these two nodes is the weight of global mincut. 

Note: The algorithm only returns the weight of mincut. I will need to find a way to recover which original vertex belongs to which cut.

### CODE REVIEW

File: src-tauri/src/globalmincut.rs [hunk 1]
```diff
@@ -0,0 +1,163 @@
+#![allow(dead_code)]
```
Comment (ajmcquilkin) [inline src-tauri/src/globalmincut.rs:1]: Why do we need this directive?

File: src-tauri/src/globalmincut.rs [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    }
+}
+
+// Create a unit test for the Graph struct
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn run_globalmincut() {
```
Comment (ajmcquilkin) [inline src-tauri/src/globalmincut.rs:109]: What are we looking for in this test case? I don't see that any conditions are being asserted

Comment (ajmcquilkin) [reply src-tauri/src/globalmincut.rs:109]: To be fair, I'm not really sure what we can assert with a probabilistic algo other than that it returns a valid node

**Review by ajmcquilkin — APPROVED** [review]
Looks solid! Just a few questions on testing and on tags

File: src-tauri/src/globalmincut.rs (hunk 2 above)
Comment (barkincavdaroglu) [reply src-tauri/src/globalmincut.rs:109]: Yeah we cannot test any of that right now. The test is to just print out the success rate.

File: src-tauri/src/globalmincut.rs (hunk 1 above)
Comment (barkincavdaroglu) [reply src-tauri/src/globalmincut.rs:1]: I was getting warnings (function x is never being used) and it bothered me lol

**Comment by barkincavdaroglu** [comment]
Since the algorithm is probabilistic, we won't always find the global mincut. For example, for the test case, we get approx. 91% success rate for finding global mincut. And some other graphs might have smaller success rates. In any case, even if the algorithm doesn't always find the global mincut, it does find the next better mincut.

File: src-tauri/src/globalmincut.rs (hunk 1 above)
Comment (ajmcquilkin) [reply src-tauri/src/globalmincut.rs:1]: Oh okay fair enough, let's keep an eye on that then into the future and make sure to remove them once they're not needed

**Pull request merged**